rocAL/rocAL_pybind/examples/rocAL_api_coco_pipeline.py

```diff
@@ -107,245 +66,162 @@ def next(self):
-            self.out, self.multiplier, self.offset, self.reverse_channels, self.tensor_format, self.tensor_dtype)
-
-        # Image id of a batch of images
-        self.loader.GetImageId(self.image_id)
-        # Image sizes of a batch
-        self.loader.GetImgSizes(self.img_size)
-        # Count of labels/ bboxes in a batch
-        self.count_batch = self.loader.GetBoundingBoxCount(
-            self.bboxes_label_count)
-        # 1D labels & bboxes array
-        if self.device == "cpu":
-          self.encoded_bboxes = np.zeros((self.count_batch*4), dtype="float32")
-          self.encoded_labels = np.zeros(self.count_batch, dtype="int32")
-          self.loader.copyEncodedBoxesAndLables(self.encoded_bboxes, self.encoded_labels)
-          encoded_bboxes_tensor = torch.tensor(self.encoded_bboxes).view(self.bs, -1, 4).contiguous()
-          encodded_labels_tensor = torch.tensor(self.encoded_labels).long().view(self.bs, -1)
         else:
-          torch_gpu_device = torch.device('cuda', self.device_id)
-          boxes_array, labels_array = self.loader.getEncodedBoxesAndLables(self.bs, int(self.num_anchors))
-          self.encoded_bboxes = torch.as_tensor(boxes_array, dtype=torch.float32, device=torch_gpu_device)
-          self.encoded_labels = torch.as_tensor(labels_array, dtype=torch.int32, device=torch_gpu_device)
-          encoded_bboxes_tensor = self.encoded_bboxes.cpu()
-          encodded_labels_tensor = self.encoded_labels.cpu()
-
-        image_id_tensor = torch.tensor(self.image_id)
-        image_size_tensor = torch.tensor(self.img_size).view(-1, self.bs, 2)
-        for i in range(self.bs):
-            index_list = []
-            actual_bboxes = []
-            actual_labels = []
-            for idx, x in enumerate(encodded_labels_tensor[i]):
-                if x != 0:
-                    index_list.append(idx)
-                    actual_bboxes.append(encoded_bboxes_tensor[i][idx].tolist())
-                    actual_labels.append(encodded_labels_tensor[i][idx].tolist())
-
-            if self.display:
-                img = self.out
-                draw_patches(img[i], self.image_id[i],
-                             actual_bboxes, self.device)
+            self.output_tensor_list = self.loader.getOutputTensors()
+        if self.out is None:
+            self.w, self.h = self.output_tensor_list[0].max_shape()
+            self.bs = self.output_tensor_list[0].batch_size()
+            self.color_format = self.output_tensor_list[0].color_format() if self.color_format else self.color_format
+            torch_gpu_device = torch.device('cuda', self.device_id)
+            if self.tensor_format == types.NCHW:
+                torch_gpu_device = torch.device('cuda', self.device_id)
+                if self.tensor_dtype == types.FLOAT:
+                    self.out = torch.empty((self.batch_size, self.color_format, self.h, self.w,), dtype = torch.float32, device = torch_gpu_device)
```

Comment:
We can get the torch dtype right?
That will reduce the lines of code please make the change